plugins/modules/azure_rm_virtualmachine.py

```diff
@@ -306,6 +315,10 @@
                 choices:
                     - ReadOnly
                     - ReadWrite
+                default: ReadOnly
```

Comment:
Argument 'caching' in argument_spec found in data_disks defines default as (None) but documentation defines default as ('ReadOnly')

Reply:
Thank you. This was a mistake. I have addressed this. New commit has been pushed.